tests/Feature/MemberProjectControllerTest.php

```diff
@@ -0,0 +1,187 @@
+<?php
+
+namespace Tests\Feature;
+
+use Tests\TestCase;
+use App\MemberProject;
+use Faker\Factory;
+use Illuminate\Foundation\Testing\WithFaker;
+use Illuminate\Foundation\Testing\RefreshDatabase;
+use Illuminate\Foundation\Testing\DatabaseMigrations;
+use Illuminate\Foundation\Testing\WithoutMiddleware;
+
+class MemberProjectControllerTest extends TestCase
+{
+    use DatabaseMigrations;
+    use WithoutMiddleware;
+    /**
+     * A basic test example.
+     *
+     * @return void
+     */
+    public function testExample()
+    {
+        $this->assertTrue(true);
+    }
+
+    public function testListMemberProjectSuccess()
+    {
+        $project = factory(MemberProject::class)->create();
+        $response = $this->get('member_projects');
+        $response->assertStatus(200);
+    }
+
+    public function testDeleteMemberProjectSuccess()
+    {
+        $array = Factory(MemberProject::class)->create()->toArray();
+        $response = $this->call('DELETE', 'member_projects/destroy', $array);
+        $this->assertEquals(200, $response->status());
+        $this->assertDatabaseMissing('projects', [
+            'meber_id'=>$array['meber_id'],
+            'project_id'=>$array['project_id'],
+            'role'=>$array['role'],
+        ]);
+    }
+
+    public function testEditMemberProjectSuccess()
+    {
+        $json = '{"id":1,"meber_id":1,"project_id":1,"role":"haha"}';
+        $array1 = [
+            'id'=>1,
+            'meber_id'=>1,
+            'project_id'=>1,
+            'role'=>'haha',
```

Comment:
'haha' is not a valid role, so this request should not be successful.

Reply:
yep